### PR TITLE
Process bpf events in parallel

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now spawn a go routine for each event to be able to process them in parallel.

#### Which issue(s) this PR fixes:

None

#### Does this PR have test?

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
Updated BPF event processing to run in parallel.
```

### CODE REVIEW

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (JAORMX) [inline internal/pkg/daemon/bpfrecorder/bpfrecorder.go:550]: While this is a good enhancement, I'm slightly concerned that we'd hit a goroutine limit in bigger deployments. Pheraps after this merges it would be good to do some stress testing of this part of the code-base.

Comment (saschagrunert) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:550]: In theory we can get 4194304 go routines for 64bit systems (max amount of pids), which could cause memory spikes. So yeah either we limit it by using a worker pool or we adjust the memory constraints accordingly.

**Comment by JAORMX** [comment]
/approve

@pjbgf @jhrozek could you also take a look?

**Review by JAORMX — APPROVED** [review]

**Comment by JAORMX** [comment]
/lgtm cancel

waiting for feedback from other maintainers

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (pjbgf) [inline internal/pkg/daemon/bpfrecorder/bpfrecorder.go:551]: Feels like something that could/should be configurable.

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (jhrozek) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:560]: Would it help, as an additional enhancement, to have a pool of gouroutines? I'm sure we're not the first ones with this requirement, so I bet there exists a library that already does this..

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (saschagrunert) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:551]: Hm, how would users decide which value to choose here? I'm not sure about making it configurable for end users without confusing them.

Comment (brness) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:551]: maybe this could be positively related with spod request and limits.

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (JAORMX) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:560]: I think @saschagrunert 's solution to use a semaphore is equivalent to having a pool. I don't have an opinion in terms of making that configurable as @pjbgf suggested, but it might be worth doing.

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go
Comment (pjbgf) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:551]: @saschagrunert the worry was mostly around giving users a way to control this in case it got out of hand.
But I am happy to ahead as-is and review before next release.

**Pull request merged**
